pulpcore/pulpcore/app/viewsets/content.py

```diff
@@ -79,3 +81,24 @@ class ArtifactViewSet(CreateDestroyReadNamedModelViewSet):
     queryset = Artifact.objects.all()
     serializer_class = ArtifactSerializer
     filter_class = ArtifactFilter
+
+    def destroy(self, request, pk):
+        """
+        Remove :class:`~pulpcore.app.models.Artifact` only if it doesn't belong to any
```

Comment:
Since artifacts don't belong to any one Content, I would suggest:

_only if it is not associated with any ~pulpcore.app.models.Content_